src/main/java/com/salesforce/storm/spout/dynamic/metrics/LogRecorder.java

```diff
@@ -60,13 +60,24 @@ public void close() {
     }
 
     @Override
-    public void count(Class sourceClass, String metricName) {
-        count(sourceClass, metricName, 1);
+    public void count(final MetricDefinition metric) {
+        countBy(metric, 1L, new Object[0]);
     }
 
     @Override
-    public void count(Class sourceClass, String metricName, long incrementBy) {
-        final String key = generateKey(sourceClass, metricName);
+    public void count(final MetricDefinition metric, final Object... metricParameters) {
+        countBy(metric, 1L, metricParameters);
+
+    }
+
+    @Override
+    public void countBy(final MetricDefinition metric, final long incrementBy) {
+        countBy(metric, incrementBy, new Object[0]);
+    }
+
+    @Override
+    public void countBy(final MetricDefinition metric, final long incrementBy, final Object... metricParameters) {
```

Comment:
@Crim I had to rename this because of ambiguity when compiling thanks to the varargs.